sound/soc/sof/ipc3-control.c

```diff
@@ -381,31 +383,46 @@ static int sof_ipc3_bytes_ext_put(struct snd_sof_control *scontrol,
 		return -EINVAL;
 	}
 
-	if (copy_from_user(cdata->data, tlvd->tlv, header.length))
-		return -EFAULT;
+	saved = kmemdup(cdata->data, scontrol->max_size, GFP_KERNEL);
```

Comment:
Add here a comment:

`` We need to restore cdata in case byte control update fails. So, save it here``